src/app/store/collectibles-metadata/reducer.ts

```diff
@@ -0,0 +1,70 @@
+import { createReducer } from '@reduxjs/toolkit';
+import { persistReducer, createTransform } from 'redux-persist';
+import hardSet from 'redux-persist/lib/stateReconciler/hardSet';
+import storage from 'redux-persist/lib/storage';
+
+import { tokenToSlug } from 'lib/assets';
+import { fromAssetSlug } from 'lib/assets/utils';
+import { checkSizeOfLocalStorageEntryToSet } from 'lib/local-storage';
+import { buildTokenMetadataFromFetched } from 'lib/metadata/utils';
+
+import {
+  putCollectiblesMetadataAction,
+  loadCollectiblesMetadataAction,
+  resetCollectiblesMetadataLoadingAction
+} from './actions';
+import { collectiblesMetadataInitialState, CollectiblesMetadataState } from './state';
+
+const collectiblesMetadataReducer = createReducer(collectiblesMetadataInitialState, builder => {
+  builder.addCase(putCollectiblesMetadataAction, (state, { payload: { records, resetLoading } }) => {
+    for (const slug of Object.keys(records)) {
```

Comment:
```suggestion
    for (const slug in records) {
```
It should work the same way